Application/EdFi.Ods.Extensions.TPDM/Artifacts/MsSql/Data/Security/1010-TPDM-ResourceClaims.sql

```diff
@@ -1,465 +1,6178 @@
-    ([Action_ActionId]
-    ,[AuthorizationStrategy_AuthorizationStrategyId]
-    ,[ResourceClaim_ResourceClaimId]
-    ,[ValidationRuleSetName])
-SELECT ac.ActionId, @AuthorizationStrategyId, ResourceClaimId, null
-FROM [dbo].[ResourceClaims]
-CROSS APPLY
-    (SELECT ActionId
-    FROM [dbo].[Actions]
-    WHERE ActionName IN ('Create', 'Read', 'Update', 'Delete')) AS ac
-WHERE ResourceName IN ('anonymizedStudent'
-, 'anonymizedStudentAssessments'
-, 'performanceMeasure'
-, 'performanceMeasureCourseAssociation'
-, 'professionalDevelopmentEvent'
-, 'recruitmentEvent'
-, 'staffTeacherPreparationProviderAssociation'
-, 'anonymizedStudentAssessment'
-, 'talentManagementGoal'
-, 'teacherCandidate'
-, 'teacherCandidateStudentGrowthMeasure'
-, 'teacherCandidateTeacherPreparationProviderAssociation'
-, 'anonymizedStudentAcademicRecord'
-, 'anonymizedStudentAssessmentCourseAssociation'
-, 'anonymizedStudentAssessmentSectionAssociation'
-, 'anonymizedStudentCourseAssociation'
-, 'anonymizedStudentCourseTranscript'
-, 'anonymizedStudentEducationOrganizationAssociation'
-, 'anonymizedStudentSectionAssociation'
-, 'applicant'
-, 'applicantProspectAssociation'
-, 'application'
-, 'applicationEvent'
-, 'completerAsStaffAssociation'
-, 'employmentEvent'
-, 'employmentSeparationEvent'
-, 'openStaffPositionEvent'
-, 'prospect'
-, 'staffApplicantAssociation'
-, 'fieldworkExperienceSectionAssociation'
-, 'staffProspectAssociation'
-, 'staffStudentGrowthMeasure'
-, 'staffStudentGrowthMeasureCourseAssociation'
-, 'staffStudentGrowthMeasureEducationOrganizationAssociation'
-, 'staffStudentGrowthMeasureSectionAssociation'
-, 'staffTeacherPreparationProviderProgramAssociation'
-, 'teacherCandidateTeacherPreparationProviderProgramAssociation'
-, 'teacherCandidateAcademicRecord'
-, 'teacherCandidateCourseTranscript'
-, 'teacherCandidateStaffAssociation'
-, 'teacherCandidateStudentGrowthMeasureCourseAssociation'
-, 'teacherCandidateStudentGrowthMeasureEducationOrganizationAssociation'
-, 'teacherCandidateStudentGrowthMeasureSectionAssociation'
-, 'teacherPreparationProviderProgram'
-, 'professionalDevelopmentEventAttendance'
-);
+BEGIN
+    DECLARE 
+        @applicationId AS INT,
+        @claimId AS INT,
+        @claimName AS nvarchar(max),
+        @parentResourceClaimId AS INT,
+        @existingParentResourceClaimId AS INT,
+        @claimSetId AS INT,
+        @claimSetName AS nvarchar(max),
+        @authorizationStrategyId AS INT,
+        @msg AS nvarchar(max),
+        -- Well-known values
+        @createActionId AS INT = 1,
+        @readActionId AS INT = 2,
+        @updateActionId AS INT = 3,
+        @deleteActionId AS INT = 4
+
+    DECLARE @claimIdStack AS TABLE (Id INT IDENTITY, ResourceClaimId INT)
+
+    SELECT @applicationId = ApplicationId
+    FROM [dbo].[Applications] WHERE ApplicationName = 'Ed-Fi ODS API';
+
+    BEGIN TRANSACTION
+
+    -- Push claimId to the stack
+    INSERT INTO @claimIdStack (ResourceClaimId) VALUES (@claimId)
```

Comment:
i am confused here, how is this temporary table being set?

Reply:
It's a table variable, declared at the top:
```sql
DECLARE @claimIdStack AS TABLE (Id INT IDENTITY, ResourceClaimId INT)
```